### PR TITLE
Doxygen doc update for heat_transfer.h

### PR DESCRIPTION
**Pull request by mivaia**
# Description of the problem

- Doxygen documentation incomplete for heat_transfer.h

# Description of the solution

- Updated it

# How Has This Been Tested?

- Web verification. 

# Comments

- First PR.
- Part of Doxygen documentation hackathon.

### CODE REVIEW

File: include/solvers/heat_transfer.h [hunk 1]
```diff
@@ -13,6 +13,14 @@
  *
  * ---------------------------------------------------------------------
 
+
+ *
+ * Implementation of heat transfer as an auxiliary physics.
+ * This heat equation is weakly coupled to the velocity field.
+ * Equation solved:
+ * rho * Cp * (dT/dt + u.gradT) = k div(gradT) + nu/rho * (gradu : gradu)
```
Comment (blaisb) [inline include/solvers/heat_transfer.h:21]:
```suggestion
 * \f$ \rho * C_p * (\frac{dT}{dt} + u \cdot \nabla T) = k \nabla^2T + \tau : \nabla u \f$
```
You should write the documentation using LaTeX format.

File: include/solvers/heat_transfer.h [hunk 2]
```diff
@@ -51,11 +59,37 @@
 #include <deal.II/lac/trilinos_vector.h>
 
 
+/**
+ * @brief Implementation of heat transfer as an auxiliary physics. The heat
+ * equation is weakly coupled to the velocity field. Equation solved:
+ * \f$ \rho C_p \cdot \left( \frac{dT}{dt} + u \cdot \nabla T \right) =
+ * k \cdot \nabla^2 T + \frac{\nu}{\rho} \cdot (\nabla u : \nabla u) \f$
```
Comment (blaisb) [inline include/solvers/heat_transfer.h:66]:
```suggestion
 * k \cdot \nabla^2 T + \tau : \nabla u \f$
```

File: include/solvers/heat_transfer.h [hunk 3]
```diff
@@ -51,11 +59,37 @@
...
 template <int dim>
 class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
 {
 public:
+  /**
+   * @brief Constructor of the HeatTransfer object
+   *
+   * @param multiphysics_interface Map of the auxiliary physics that will be
+   * solved on top of a computational fluid dynamic simulation. The keys are
+   * the Parameters::PhysicsID int enum.
```
Comment (blaisb) [inline include/solvers/heat_transfer.h:79]:
```suggestion
   * @param multiphysics_interface Map of the auxiliary physics that will be
   * solved on top of a computational fluid dynamic simulation. 
```

File: include/solvers/heat_transfer.h [hunk 4]
```diff
@@ -125,38 +159,41 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
...
   assemble_rhs();
 
   /**
    * @brief Call for the assembly of the matrix and the right-hand side of the Nitsche restriction for the heat transfert equation.
    *
-   * @param assemble_matrix boolean that is true for matrix assembly, and false for rhs assembly
+   * @param assemble_matrix boolean that is true for matrix assembly, and false for rhs assembly.
    */
   void
   assemble_nitsche_heat_restriction(bool assemble_matrix);
```
Comment (blaisb) [inline include/solvers/heat_transfer.h:182]:
```suggestion
  assemble_nitsche_heat_restriction(const bool assemble_matrix);
```
Good opportunity to make this a const :)

File: include/solvers/heat_transfer.h [hunk 5]
```diff
@@ -186,110 +223,162 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
...
   get_dof_handler() override
   {
     return dof_handler;
   }
+
+  /**
+   * @brief Getter method to access the private attribute evaluation_point for
+   * the physic currently solved.
+   *
+   * @return The point at which the evaluation is performed.
```
Comment (blaisb) [inline include/solvers/heat_transfer.h:313]:
```suggestion
   * @return The vector at which the evaluation is performed.
```

File: include/solvers/heat_transfer.h [hunk 6]
```diff
@@ -186,110 +223,162 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
...
   {
     return local_evaluation_point;
   }
+
+  /**
+   * @brief Getter method to access the private attribute
+   * newton_update for the physic currently solved.
+   *
+   * @return The direction \f$ \delta u^n \f$ used to perform the newton iteration
+   * in \f$ u^{n+1} = u^n + \alpha ^n \delta u ^n \f$. Where \f$ \alpha ^n \f$
```
Comment (blaisb) [inline include/solvers/heat_transfer.h:339]: That's maybe a lot of information. The Newton solver should contain the details regarding the Newton update itself :)

File: include/solvers/heat_transfer.h [hunk 7]
```diff
@@ -446,19 +535,20 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
...
-   * @param gather_vof boolean true when VOF=true (multiphase flow), used to gather
-   * VOF information
+   * @param gather_vof Boolean true when VOF=true (multiphase flow), used to gather
+   * VOF information.
    *
    * @param monitored_fluid Fluid indicator (fluid0 or fluid1 or both) corresponding
    * to the phase of interest.
    *
-   * @param domain_name string indicating the monitored_fluid in the console output,
+   * @param domain_name Btring indicating the monitored_fluid in the console output,
```
Comment (blaisb) [inline include/solvers/heat_transfer.h:547]:
```suggestion
   * @param domain_name String indicating the monitored_fluid in the console output,
```

File: include/solvers/heat_transfer.h [hunk 8]
```diff
@@ -506,74 +596,191 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
...
   std::vector<GlobalVectorType> previous_solutions;
 
-  // Solution transfer classes
+  /**
+   * @brief A parallel::distributed::SolutionTransfer object implements the
+   * transfer of a discrete FE function (e.g. a solution vector) from one mesh
+   * to another that is obtained from the first by a single refinement and/or
+   * coarsening step. During interpolation the vector is reinitialized to the
+   * new size and filled with the interpolated values. This Deal.ii class is
+   * used for mesh_refinement and simulation restarts.
```
Comment (blaisb) [inline include/solvers/heat_transfer.h:722]: I don't think this needs this much detail here, but if you feel like it does, why not

File: include/solvers/heat_transfer.h [hunk 9]
```diff
@@ -506,74 +596,191 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
...
+
+  /**
+   * @brief A parallel::distributed::SolutionTransfer object implements the
+   * transfer of a discrete FE function (e.g. a solution vector) from one mesh
+   * to another that is obtained from the first by a single refinement and/or
+   * coarsening step. During interpolation the vector is reinitialized to the
+   * new size and filled with the interpolated values. This Deal.ii class is
+   * used for mesh_refinement and simulation restarts.
+   * previous_solutions_transfer occupies the same role as solution_transfer
+   * but for solutions at previous time steps.
```
Comment (blaisb) [inline include/solvers/heat_transfer.h:736]: You already said it above, so here I would not repeat myself.

**Review by blaisb — CHANGES_REQUESTED** [review]
Some small changes. Nothing big, but some of these comments are very verbose lets say :)

File: include/solvers/heat_transfer.h [hunk 10]
```diff
@@ -51,11 +59,36 @@
...
+   * @param p_simulation_parameters Contain the simulation parameter file
+   * information.
+   *
+   * @param p_triangulation Contain the mesh information. In a
+   * parallel::DistributedTriangulationBase<dim> not every detail may be known
+   * on each processor. The mesh is distributed between the processors.
+   *
+   * @param p_simulation_control Object responsible for the control of
+   * steady-state and transient simulations. Contains all the information
+   * related to time stepping and the stopping criterions.
```
Comment (oguevremont) [inline include/solvers/heat_transfer.h:89]:
```suggestion
   * related to time stepping and the stopping criteria.
```

File: include/solvers/heat_transfer.h [hunk 11]
```diff
@@ -125,38 +158,41 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
...
+   * is integrated to this solver.
    */
   void
   assemble_rhs();
 
   /**
    * @brief Call for the assembly of the matrix and the right-hand side of the Nitsche restriction for the heat transfert equation.
    *
-   * @param assemble_matrix boolean that is true for matrix assembly, and false for rhs assembly
+   * @param assemble_matrix boolean that is true for matrix assembly, and false for rhs assembly.
```
Comment (oguevremont) [inline include/solvers/heat_transfer.h:178]:
```suggestion
   * @param assemble_matrix Boolean that is true for matrix assembly, and false for rhs assembly.
```

File: include/solvers/heat_transfer.h [hunk 12]
```diff
@@ -186,110 +222,160 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
...
   /**
-   * @brief Prepares Heat Transfer to write checkpoint
+   * @brief Prepare Heat Transfer to write checkpoint
    */
   void
   write_checkpoint() override;
 
   /**
-   * @brief Allows Heat Transfer to set-up solution vector from checkpoint file;
+   * @brief Allow Heat Transfer to set-up solution vector from checkpoint file.
```
Comment (oguevremont) [inline include/solvers/heat_transfer.h:256]:
```suggestion
   * @brief Allow Heat Transfer to set up solution vector from checkpoint file.
```

File: include/solvers/heat_transfer.h [hunk 13]
```diff
@@ -186,110 +222,160 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
...
    */
   void
   read_checkpoint() override;
 
   /**
-   * @brief Sets-up the DofHandler and the degree of freedom associated with the physics.
+   * @brief Set-up the DofHandler and the degree of freedom associated with the physics.
```
Comment (oguevremont) [inline include/solvers/heat_transfer.h:262]:
```suggestion
   * @brief Set up the DofHandler and the degrees of freedom associated with the physics.
```

File: include/solvers/heat_transfer.h [hunk 14]
```diff
@@ -186,110 +222,160 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
...
    */
   void
   setup_dofs() override;
 
   /**
-   * @brief Sets-up the initial conditions associated with the physics. Generally, physics
-   * only support imposing nodal values, but some physics additionnaly support
-   * the use of L2 projection or steady-state solutions.
+   * @brief Set-up the initial conditions associated with the physics.
```
Comment (oguevremont) [inline include/solvers/heat_transfer.h:268]:
```suggestion
   * @brief Set up the initial conditions associated with the physics.
```

File: include/solvers/heat_transfer.h [hunk 15]
```diff
@@ -353,7 +439,7 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
     StabilizedMethodsCopyData                            &copy_data);
 
   /**
-   * @brief sets up the vector of assembler functions
+   * @brief Set-up the vector of assembler functions
```
Comment (oguevremont) [inline include/solvers/heat_transfer.h:442]:
```suggestion
   * @brief Set up the vector of assembler functions
```

File: include/solvers/heat_transfer.h [hunk 16]
```diff
@@ -186,110 +222,160 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
...
+   * @param estimated_error_per_cell The deal.II vector of estimated_error_per_cell.
    */
   void
   compute_kelly(const std::pair<const Parameters::MeshAdaptation::Variable,
                                 Parameters::MultipleAdaptationParameters> &ivar,
                 dealii::Vector<float> &estimated_error_per_cell) override;
 
   /**
-   * @brief Prepares Heat Transfer to write checkpoint
+   * @brief Prepare Heat Transfer to write checkpoint
```
Comment (oguevremont) [inline include/solvers/heat_transfer.h:250]:
```suggestion
   * @brief Prepare Heat Transfer to write checkpoint.
```

File: include/solvers/heat_transfer.h [hunk 17]
```diff
@@ -186,110 +222,160 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
...
    */
   void
   setup_dofs() override;
 
   /**
-   * @brief Sets-up the initial conditions associated with the physics. Generally, physics
-   * only support imposing nodal values, but some physics additionnaly support
-   * the use of L2 projection or steady-state solutions.
+   * @brief Set-up the initial conditions associated with the physics.
+   * Generally, physics only support imposing nodal values, but some physics
```
Comment (oguevremont) [inline include/solvers/heat_transfer.h:269]: I would put this comment in Auxiliary Physics and write specifically what Heat Transfer allows as initial conditions.

File: include/solvers/heat_transfer.h [hunk 18]
```diff
@@ -186,110 +222,160 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
...
+   * @brief Update non zero constraints if the boundary is time-dependent.
    */
   void
   update_boundary_conditions() override;
 
   /**
-   * @brief Call for the solution of the linear system of equation using a strategy appropriate
-   * to the auxiliary physics
+   * @brief Call for the solution of the linear system of equation using a
+   * strategy appropriate to the auxiliary physics.
```
Comment (oguevremont) [inline include/solvers/heat_transfer.h:283]: Same comment here for specificity, if you agree.

File: include/solvers/heat_transfer.h [hunk 19]
```diff
@@ -186,110 +222,160 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
...
   get_present_solution() override
   {
     return present_solution;
   }
+
+  /**
+   * @brief Getter method to access the private attribute
+   * system_rhs for the physic currently solved.
+   *
+   * @return Right hand side vector
```
Comment (oguevremont) [inline include/solvers/heat_transfer.h:363]:
```suggestion
   * @return Right hand side vector.
```

File: include/solvers/heat_transfer.h [hunk 20]
```diff
@@ -416,7 +502,7 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
   postprocess_liquid_fraction(const bool gather_vof);
 
   /**
-   * @brief Post-processing. Write the liquid fraction to a file
+   * @brief Post-processing. Write the liquid fraction to an output file
```
Comment (oguevremont) [inline include/solvers/heat_transfer.h:505]:
```suggestion
   * @brief Post-processing. Write the liquid fraction to an output file.
```

File: include/solvers/heat_transfer.h [hunk 21]
```diff
@@ -506,74 +593,183 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
...
+
+  /**
+   * @brief Collection of cells that cover the domain on which one wants to
+   * solve a partial differential equation.
+   */
   std::shared_ptr<parallel::DistributedTriangulationBase<dim>> triangulation;
+  /**
+   * @brief Responsible for the control of steady-state and transient
+   * simulations. Contains all the information related to time stepping and the
+   * stopping criterion. See simulation_control abstract class for more
```
Comment (oguevremont) [inline include/solvers/heat_transfer.h:618]:
```suggestion
   * stopping criteria. See simulation_control abstract class for more
```

File: include/solvers/heat_transfer.h [hunk 22]
```diff
@@ -506,74 +593,183 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
...
+   * Evaluate convergence rates or orders.
+   */
+  ConvergenceTable error_table;
 
   // Mapping and Quadrature
-  std::shared_ptr<Mapping<dim>>        temperature_mapping;
-  std::shared_ptr<Quadrature<dim>>     cell_quadrature;
+
+  /**
+   * @brief Is a transformation which maps point in the reference cell to
```
Comment (oguevremont) [inline include/solvers/heat_transfer.h:643]:
```suggestion
   * @brief Transformation which maps point in the reference cell to
```

File: include/solvers/heat_transfer.h [hunk 23]
```diff
@@ -506,74 +593,183 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
...
+   * points in the actual grid cell.
+   */
+  std::shared_ptr<Mapping<dim>> temperature_mapping;
+  /**
+   * @brief Approximate an integral by evaluating the integrand at specific
+   * points and summing the point values with specific weights
```
Comment (oguevremont) [inline include/solvers/heat_transfer.h:649]:
```suggestion
   * points and summing the point values with specific weights.
```

File: include/solvers/heat_transfer.h [hunk 24]
```diff
@@ -506,74 +593,183 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
...
+   */
+  std::shared_ptr<Quadrature<dim>> cell_quadrature;
+  /**
+   * @brief Approximate an integral by evaluating the integrand at specific
+   * points and summing the point values with specific weights
```
Comment (oguevremont) [inline include/solvers/heat_transfer.h:654]:
```suggestion
   * points and summing the point values with specific weights.
```

File: include/solvers/heat_transfer.h [hunk 25]
```diff
@@ -506,74 +593,183 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
...
   // Solution storage:
+
+  /**
+   * @brief Store a subset of the DoF indices that are stored on a
+   * particular processor on a distributed architecture.
+   */
   IndexSet locally_owned_dofs;
+  /**
+   * @brief Store a subset of the DoF indices that contains both the
+   * locally_owned_dofs and the DoF indices on all ghost cells.
```
Comment (oguevremont) [inline include/solvers/heat_transfer.h:668]:
```suggestion
   * locally_owned_dofs and the DoF indices on ghost cells.
```

**Review by oguevremont — COMMENTED** [review]
Nice PR!

**Comment by blaisb** [comment]
Is this ready for a re-review?

**Comment by mivaia** [comment]
> Is this ready for a re-review?

Yess it is!

File: include/solvers/heat_transfer.h [hunk 26]
```diff
@@ -51,11 +60,36 @@
...
+ * k \cdot \nabla^2 T + \tau : \nabla u \f$
+ *
+ */
 
 template <int dim>
 class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
 {
 public:
+  /**
+   * @brief Constructor of the HeatTransfer object
```
Comment (oguevremont) [inline include/solvers/heat_transfer.h:76]:
```suggestion
   * @brief Constructor of the HeatTransfer object.
```

**Review by oguevremont — APPROVED** [review]

**Review by blaisb — APPROVED** [review]

**Pull request merged**